model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/authorization/PolicyAdapter.java

```diff
@@ -58,9 +58,9 @@ public PolicyAdapter(CachedPolicy cached, StoreFactoryCacheSession cacheSession)
     public Policy getDelegateForUpdate() {
         if (updated == null) {
             updated = modelSupplier.get();
+            if (updated == null) throw new IllegalStateException("Not found in database");
             String defaultResourceType = updated.getConfig().get("defaultResourceType");
             cacheSession.registerPolicyInvalidation(cached.getId(), cached.getName(), cached.getResourcesIds(modelSupplier), cached.getScopesIds(modelSupplier), defaultResourceType, cached.getResourceServerId());
-            if (updated == null) throw new IllegalStateException("Not found in database");
```

Comment:
Nitpick: add curly braces for the statement after the `if`

Reply:
I'm fine with that! Didn't even think about it since I just moved the line 😆